pkg/processor/trigger/trigger.go

```diff
@@ -83,6 +86,7 @@ type Trigger interface {
 
 // AbstractTrigger implements common trigger operations
 type AbstractTrigger struct {
+	Trigger
```

Comment:
why anonymously structure  embedding? you may want to use `Trigger *Trigger` to 
a. keep the structure lightweight
b. abstract is abstract, it holds common fields but not the inheritor's ones
c. point vs non-pointer because why allocating twice the size?
d. not anonymously to avoid collisions